### PR TITLE
Organize unit tests

### PR DESCRIPTION
**Pull request by bandinib-amzn**
*Issue #, if available:*

*Description of changes:*

- Created folder named `test` under each directory and moved all unit tests to `test` folder.
- Fixed broken unit tests
- Updated snapshots.
- Added unit test for permission tree.

*Testing:*

- Ran `yarn tsc`
- Ran `yarn test:jest_ui --coverage `

*Result of testing:*
```
opendistro_security % yarn test:jest_ui --coverage  
yarn run v1.22.4
$ node ./test/run_jest_tests.js --config ./test/jest.config.ui.js --coverage
 PASS  public/apps/login/test/login-page.test.tsx
 PASS  public/apps/configuration/panels/tenant-list/test/tenant-list.test.tsx
 PASS  public/apps/configuration/panels/role-edit/test/role-edit.test.tsx
 PASS  public/apps/configuration/panels/test/role-list.test.tsx
 PASS  public/apps/configuration/panels/internal-user-edit/test/internal-user-edit.test.tsx
 PASS  public/apps/configuration/panels/role-edit/test/index-permission-panel.test.tsx
 PASS  public/apps/configuration/panels/role-view/test/index-permission-panel.test.tsx
 PASS  public/apps/configuration/panels/permission-list/test/permission-list.test.tsx
 PASS  public/apps/configuration/panels/audit-logging/test/audit-logging.test.tsx
 PASS  public/apps/configuration/panels/role-view/test/tenants-panel.test.tsx
 PASS  public/apps/configuration/panels/role-mapping/test/role-edit-mapped-user.test.tsx
 PASS  public/apps/configuration/panels/test/user-list.test.tsx
 PASS  public/apps/configuration/panels/auth-view/test/instruction-view.test.tsx
 PASS  public/apps/configuration/panels/role-view/test/role-view.test.tsx
 PASS  public/apps/account/test/tenant-switch-panel.test.tsx
 PASS  public/apps/configuration/panels/internal-user-edit/test/attribute-panel.test.tsx
 PASS  public/apps/account/test/account-nav-button.test.tsx
 PASS  public/apps/configuration/panels/role-edit/test/cluster-permission-panel.test.tsx
 PASS  public/apps/configuration/panels/role-edit/test/tenant-panel.test.tsx
 PASS  public/apps/configuration/panels/auth-view/test/authentication-sequence-panel.test.tsx
 PASS  public/apps/configuration/panels/auth-view/test/auth-view.test.tsx
 PASS  public/apps/configuration/panels/audit-logging/test/code-editor.test.tsx
 PASS  public/apps/configuration/panels/audit-logging/test/edit-setting-group.test.tsx
 PASS  public/apps/configuration/utils/test/display-utils.test.tsx
 PASS  public/apps/configuration/panels/tenant-list/test/edit-modal.test.tsx
 PASS  public/apps/configuration/utils/test/password-edit-panel.test.tsx
 PASS  public/apps/configuration/panels/permission-list/test/edit-modal.test.tsx
 PASS  public/apps/configuration/panels/role-mapping/test/external-identities-panel.test.tsx
 PASS  public/apps/configuration/panels/test/get-started.test.tsx
 PASS  public/apps/configuration/utils/test/panel-with-header.test.tsx
 PASS  public/apps/account/test/log-out-button.test.tsx
 PASS  public/apps/configuration/panels/permission-tree/test/permission-tree.test.tsx
 PASS  public/apps/configuration/panels/audit-logging/test/audit-logging-edit-settings.test.tsx
 PASS  public/apps/configuration/utils/test/form-row.test.tsx
 PASS  public/apps/configuration/panels/role-view/test/cluster-permission-panel.test.tsx
 PASS  public/apps/configuration/utils/test/delete-confirm-modal-utils.test.tsx
 PASS  public/apps/configuration/utils/test/name-row.test.tsx
 PASS  public/apps/configuration/panels/role-mapping/test/users-panel.test.tsx
 PASS  public/apps/account/test/password-reset-panel.test.tsx
 PASS  public/apps/configuration/panels/audit-logging/test/view-setting-group.test.tsx
 PASS  public/apps/configuration/panels/auth-view/test/authorization-panel.test.tsx
 PASS  public/apps/account/test/role-info-panel.test.tsx
 PASS  public/apps/configuration/utils/test/toast-utils.test.tsx
 PASS  public/apps/configuration/utils/test/resource-validation-util.test.tsx
 PASS  public/apps/configuration/utils/test/index-permission-utils.test.tsx
 PASS  public/apps/configuration/utils/test/tenant-utils.test.tsx
 PASS  public/apps/configuration/utils/test/role-list-utils.test.tsx
 PASS  public/apps/configuration/utils/test/role-mapping-utils.test.tsx
 PASS  public/apps/configuration/utils/test/action-groups-utils.test.tsx
 PASS  public/apps/configuration/utils/test/internal-user-list-utils.test.tsx
 PASS  public/apps/configuration/utils/test/resource-utils.test.tsx
-----------------------------------------------------|---------|----------|---------|---------|----------------------
File                                                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s    
-----------------------------------------------------|---------|----------|---------|---------|----------------------
All files                                            |   67.81 |    67.74 |   54.47 |   68.35 |                      
 public                                              |       0 |        0 |       0 |       0 |                      
  index.ts                                           |       0 |      100 |       0 |       0 | 24                   
  plugin.ts                                          |       0 |        0 |       0 |       0 | 47-140               
  types.ts                                           |       0 |        0 |       0 |       0 |                      
 public/apps                                         |    87.5 |       75 |     100 |    87.5 |                      
  apps-constants.tsx                                 |     100 |      100 |     100 |     100 |                      
  error-utils.ts                                     |      80 |       75 |     100 |      80 | 29                   
  translation-utils.ts                               |     100 |      100 |     100 |     100 |                      
  types.ts                                           |       0 |        0 |       0 |       0 |                      
 public/apps/account                                 |   60.84 |    48.53 |   38.78 |   62.73 |                      
  account-app.tsx                                    |       0 |        0 |       0 |       0 | 24-40                
  account-nav-button.tsx                             |   35.29 |       75 |    9.09 |      40 | 78-111,132-140       
  constants.tsx                                      |     100 |      100 |     100 |     100 |                      
  log-out-button.tsx                                 |     100 |      100 |     100 |     100 |                      
  password-reset-panel.tsx                           |     100 |       50 |     100 |     100 | 140                  
  role-info-panel.tsx                                |   76.92 |      100 |      60 |   76.92 | 38,59-73             
  tenant-selection-page.tsx                          |       0 |        0 |       0 |       0 | 24-56                
  tenant-switch-panel.tsx                            |   67.11 |     47.5 |   61.54 |   67.11 | ...5,168,189,198-216 
  types.ts                                           |       0 |        0 |       0 |       0 |                      
  utils.tsx                                          |       0 |        0 |       0 |       0 | 23-53                
 public/apps/configuration                           |   29.33 |        0 |       0 |   30.14 |                      
  app-router.tsx                                     |       0 |        0 |       0 |       0 | 42-235               
  configuration-app.tsx                              |       0 |      100 |       0 |       0 | 31-38                
  constants.tsx                                      |      84 |        0 |       0 |      84 | 100-101,171-172      
  cross-page-toast.tsx                               |       0 |        0 |       0 |       0 | 24-34                
  types.ts                                           |       0 |        0 |       0 |       0 |                      
  ui-constants.tsx                                   |     100 |      100 |     100 |     100 |                      
 public/apps/configuration/panels                    |   71.57 |    65.38 |   48.72 |   74.49 |                      
  expression-modal.tsx                               |    62.5 |       50 |   33.33 |   83.33 | 37                   
  get-started.tsx                                    |   88.89 |      100 |    87.5 |   88.89 | 233                  
  nav-panel.tsx                                      |       0 |      100 |       0 |       0 | 21-36                
  role-list.tsx                                      |   60.98 |       60 |      25 |    62.5 | ...9,152-176,187-190 
  user-list.tsx                                      |   85.37 |    71.43 |   61.54 |    87.5 | 144-179,238-239      
 public/apps/configuration/panels/audit-logging      |   74.23 |    66.04 |      50 |   74.23 |                      
  audit-logging-edit-settings.tsx                    |   51.16 |    33.33 |   46.15 |   51.16 | ...,69,83-94,106-138 
  audit-logging.tsx                                  |      56 |       75 |   44.44 |      56 | ...0,138-146,171-210 
  code-editor.tsx                                    |     100 |      100 |     100 |     100 |                      
  constants.tsx                                      |   94.29 |        0 |       0 |   94.29 | 80,111               
  edit-setting-group.tsx                             |   76.47 |    81.82 |   45.45 |   76.47 | ...5,111-112,117,124 
  types.tsx                                          |       0 |        0 |       0 |       0 |                      
  view-setting-group.tsx                             |     100 |       75 |     100 |     100 | 36,40,44             
 public/apps/configuration/panels/auth-view          |   84.62 |    93.75 |   61.54 |   84.62 |                      
  auth-view.tsx                                      |   93.75 |       50 |     100 |   93.75 | 54                   
  authentication-sequence-panel.tsx                  |   78.95 |      100 |      40 |   78.95 | 53-62,102-103        
  authorization-panel.tsx                            |   81.25 |      100 |      50 |   81.25 | 44,88-89             
  instruction-view.tsx                               |     100 |      100 |     100 |     100 |                      
 public/apps/configuration/panels/internal-user-edit |   87.27 |    83.33 |   89.47 |   88.68 |                      
  attribute-panel.tsx                                |     100 |      100 |     100 |     100 |                      
  internal-user-edit.tsx                             |   81.08 |       70 |   66.67 |   83.33 | ...4,115,120-121,167 
  types.tsx                                          |       0 |        0 |       0 |       0 |                      
 public/apps/configuration/panels/permission-list    |   82.43 |     87.5 |   61.54 |   83.56 |                      
  edit-modal.tsx                                     |     100 |       50 |     100 |     100 | 100                  
  permission-list.tsx                                |   80.88 |    90.91 |   58.33 |   82.09 | ...6,304,314,374-375 
 public/apps/configuration/panels/permission-tree    |     100 |      100 |     100 |     100 |                      
  index.ts                                           |       0 |        0 |       0 |       0 |                      
  permission-tree.tsx                                |     100 |      100 |     100 |     100 |                      
 public/apps/configuration/panels/role-edit          |   87.85 |    78.57 |   89.13 |    87.5 |                      
  cluster-permission-panel.tsx                       |     100 |      100 |     100 |     100 |                      
  constant.tsx                                       |     100 |      100 |     100 |     100 |                      
  index-permission-panel.tsx                         |   93.75 |    83.33 |   90.91 |    93.1 | 272-290              
  role-edit.tsx                                      |   78.85 |    66.67 |   72.73 |   78.85 | ...2,145,162-163,235 
  tenant-panel.tsx                                   |     100 |      100 |     100 |     100 |                      
  types.tsx                                          |       0 |        0 |       0 |       0 |                      
 public/apps/configuration/panels/role-mapping       |   87.72 |       90 |      90 |    87.5 |                      
  external-identities-panel.tsx                      |     100 |      100 |     100 |     100 |                      
  role-edit-mapped-user.tsx                          |   82.05 |       75 |      75 |   82.05 | ...89-90,102,127,163 
  types.tsx                                          |       0 |        0 |       0 |       0 |                      
  users-panel.tsx                                    |     100 |      100 |     100 |     100 |                      
 public/apps/configuration/panels/role-view          |   69.12 |    57.14 |      50 |   69.63 |                      
  cluster-permission-panel.tsx                       |      75 |       75 |      50 |      75 | 43                   
  index-permission-panel.tsx                         |   65.38 |    72.22 |      75 |   65.38 | 53,70,131-142,191    
  role-view.tsx                                      |    80.3 |       50 |   41.67 |   81.54 | ...3,194,303,360,370 
  tenants-panel.tsx                                  |    52.5 |       25 |      30 |    52.5 | ...9-107,134-161,177 
 public/apps/configuration/panels/tenant-list        |   61.39 |    68.75 |   31.25 |   62.63 |                      
  edit-modal.tsx                                     |    87.5 |       50 |   66.67 |    87.5 | 55                   
  tenant-instruction-view.tsx                        |       0 |      100 |       0 |       0 | 20                   
  tenant-list.tsx                                    |   59.78 |    71.43 |   28.57 |   61.11 | ...4,301,324,382-383 
 public/apps/configuration/utils                     |   65.32 |    81.68 |   59.02 |   63.91 |                      
  action-groups-utils.tsx                            |   14.29 |    83.33 |   16.67 |    9.09 | 31-32,49-98          
  array-state-utils.tsx                              |    4.17 |        0 |       0 |    4.17 | 20-43,73-120         
  audit-logging-utils.tsx                            |       0 |      100 |       0 |       0 | 21-28                
  auth-view-utils.tsx                                |       0 |      100 |       0 |       0 | 20-21                
  combo-box-utils.tsx                                |      75 |      100 |   66.67 |      75 | 61                   
  context-menu.tsx                                   |   88.89 |      100 |      75 |    87.5 | 40                   
  delete-confirm-modal-utils.tsx                     |   72.73 |       75 |      50 |   77.78 | 35-36                
  display-utils.tsx                                  |   78.95 |    68.42 |   83.33 |   78.95 | 82-96                
  form-row.tsx                                       |     100 |      100 |     100 |     100 |                      
  index-permission-utils.tsx                         |     100 |      100 |     100 |     100 |                      
  internal-user-detail-utils.tsx                     |       0 |      100 |       0 |       0 | 21-29                
  internal-user-list-utils.tsx                       |      25 |       50 |   33.33 |   14.29 | 34-49                
  loading-spinner-utils.tsx                          |     100 |      100 |     100 |     100 |                      
  name-row.tsx                                       |     100 |      100 |     100 |     100 |                      
  panel-with-header.tsx                              |     100 |      100 |     100 |     100 |                      
  password-edit-panel.tsx                            |     100 |      100 |     100 |     100 |                      
  request-utils.ts                                   |       0 |        0 |       0 |       0 | 29-51                
  resource-utils.tsx                                 |     100 |      100 |     100 |     100 |                      
  resource-validation-util.tsx                       |     100 |      100 |     100 |     100 |                      
  role-detail-utils.tsx                              |       0 |      100 |       0 |       0 | 21-25                
  role-list-utils.tsx                                |   44.44 |       70 |   57.14 |    37.5 | 95-108               
  role-mapping-utils.tsx                             |   71.43 |      100 |      60 |      60 | 33,55                
  storage-utils.tsx                                  |      30 |        0 |      50 |      30 | 24-28,42-44          
  tenant-utils.tsx                                   |   84.44 |    95.24 |   64.71 |   82.93 | 50-54,81-101         
  toast-utils.tsx                                    |     100 |      100 |     100 |     100 |                      
  url-builder.tsx                                    |     100 |      100 |     100 |     100 |                      
 public/apps/login                                   |   64.52 |       75 |   66.67 |   66.67 |                      
  login-app.tsx                                      |       0 |      100 |       0 |       0 | 29-30                
  login-page.tsx                                     |   71.43 |       75 |     100 |   71.43 | 50,68-69,73-74,83-85 
 public/services                                     |       0 |        0 |       0 |       0 |                      
  chrome_wrapper.ts                                  |       0 |        0 |       0 |       0 | 19-21                
 public/utils                                        |       0 |      100 |       0 |       0 |                      
  auth-info-utils.tsx                                |       0 |      100 |       0 |       0 | 21-25                
  login-utils.tsx                                    |       0 |      100 |       0 |       0 | 23                   
-----------------------------------------------------|---------|----------|---------|---------|----------------------

Test Suites: 51 passed, 51 total
Tests:       225 passed, 225 total
Snapshots:   18 passed, 18 total
Time:        48.59s, estimated 49s
Ran all test suites.
✨  Done in 54.70s.  
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/apps/account/test/__snapshots__/log-out-button.test.tsx.snap [hunk 1]
```diff
@@ -5,7 +5,7 @@ exports[`Account menu - Log out button renders renders when auth type is OpenId
   <EuiButtonEmpty
     color="danger"
     data-test-subj="log-out-1"
-    href="/app/kibana/auth/logout"
+    href="/auth/logout"
```
Comment (tianleh) [inline public/apps/account/test/__snapshots__/log-out-button.test.tsx.snap:8]: This does not look like pure organization change. Is this intentional?

File: public/apps/configuration/panels/test/__snapshots__/get-started.test.tsx.snap [hunk 2]
```diff
@@ -238,6 +238,35 @@ exports[`Get started (landing page) renders 1`] = `
         </EuiButton>
       </EuiText>
     </EuiPanel>
+    <EuiSpacer
```
Comment (tianleh) [inline public/apps/configuration/panels/test/__snapshots__/get-started.test.tsx.snap:241]: same here

**Review by tianleh — COMMENTED** [review]
Thanks for the refactoring

File: public/apps/account/test/__snapshots__/log-out-button.test.tsx.snap (hunk 1 above)
Comment (bandinib-amzn) [reply public/apps/account/test/__snapshots__/log-out-button.test.tsx.snap:8]: Yes. This is updated snapshot. I have also updated description.

File: public/apps/configuration/panels/test/__snapshots__/get-started.test.tsx.snap (hunk 2 above)
Comment (bandinib-amzn) [reply public/apps/configuration/panels/test/__snapshots__/get-started.test.tsx.snap:241]: This is updated snapshot.

**Review by tianleh — APPROVED** [review]

**Pull request merged**